apps/ex_wire/lib/ex_wire/packet/snapshot_data/block_chunk.ex

```diff
@@ -0,0 +1,292 @@
+defmodule ExWire.Packet.SnapshotData.BlockChunk do
+  @moduledoc """
+  Block chunks contain raw block data: blocks themselves, and their transaction
+  receipts. The blocks are stored in the “abridged block” format (referred to
+  by AB), and the the receipts are stored in a list: [`receipt_1`: `P`,
+  `receipt_2`: `P`, ...] (referred to by RC).
+  """
+
+  defmodule BlockHeader do
+    @type t :: %__MODULE__{
+            # Header fields
+            author: EVM.address(),
+            state_root: EVM.hash(),
+            logs_bloom: <<_::2048>>,
+            difficulty: integer(),
+            gas_limit: integer(),
+            gas_used: integer(),
+            timestamp: integer(),
+            extra_data: integer(),
+
+            # Ommers and transactions inline
+            transactions: list(Blockchain.Transaction.t()),
+            transactions_rlp: list(ExRLP.t()),
+            ommers: list(Block.Header.t()),
+            ommers_rlp: list(ExRLP.t()),
+
+            # Seal fields
+            mix_hash: EVM.hash(),
+            nonce: <<_::64>>
+          }
+
+    defstruct [
+      :author,
+      :state_root,
+      :logs_bloom,
+      :difficulty,
+      :gas_limit,
+      :gas_used,
+      :timestamp,
+      :extra_data,
+      :transactions,
+      :transactions_rlp,
+      :ommers,
+      :ommers_rlp,
+      :mix_hash,
+      :nonce
+    ]
+  end
+
+  defmodule BlockData do
+    @type t :: %__MODULE__{
+            header: BlockHeader.t(),
+            receipts: list(Blockchain.Transaction.Receipt.t()),
+            receipts_rlp: list(ExRLP.t())
+          }
+    defstruct [
+      :header,
+      :receipts,
+      :receipts_rlp
+    ]
+  end
+
+  @type t :: %__MODULE__{
```

Comment:
Would prefer if these were up by the `defmodule` so it's easy to see what `%__MODULE__` is referring to. Unless this is pretty standard in Elixir, in which case I'll just have to get used to it =D